packages/ember-glimmer/tests/integration/helpers/input-test.js

```diff
@@ -0,0 +1,448 @@
+    }}`, {
+      disabled: false,
+      name: 'original-name',
+      checked: false,
+      tabindex: 10
+    });
+
+    this.assertSingleCheckbox();
+    this.assertNotDisabled();
+    this.assertAttr('name', 'original-name');
+    this.assertAttr('tabindex', '10');
+
+    this.runTask(() => this.rerender());
+
+    this.assertSingleCheckbox();
+    this.assertNotDisabled();
+    this.assertAttr('name', 'original-name');
+    this.assertAttr('tabindex', '10');
+
+    this.runTask(() => {
+      set(this.context, 'disabled', true);
+      set(this.context, 'name', 'updated-name');
+      set(this.context, 'tabindex', 11);
+    });
+
+    this.assertSingleCheckbox();
+    this.assertDisabled();
+    this.assertAttr('name', 'updated-name');
+    this.assertAttr('tabindex', '11');
+
+    this.runTask(() => {
+      set(this.context, 'disabled', false);
+      set(this.context, 'name', 'original-name');
+      set(this.context, 'tabindex', 10);
+    });
+
+    this.assertSingleCheckbox();
+    this.assertNotDisabled();
+    this.assertAttr('name', 'original-name');
+    this.assertAttr('tabindex', '10');
+  }
+
+  ['@test `value` property assertion']() {
+    expectAssertion(() => {
+      this.render(`{{input type="checkbox" value=value}}`, { value: 'value' });
+    }, /you must use `checked=/);
+  }
+
+  ['@test with a bound type'](assert) {
+    this.render(`{{input type=inputType checked=isChecked}}`, { inputType: 'checkbox', isChecked: true });
+
+    this.assertSingleCheckbox();
+    this.assertCheckboxIsChecked();
+
+    this.runTask(() => this.rerender());
+
+    this.assertCheckboxIsChecked();
+
+    this.runTask(() => set(this.context, 'isChecked', false));
+
+    this.assertCheckboxIsNotChecked();
+
+    this.runTask(() => set(this.context, 'isChecked', true));
+
+    this.assertCheckboxIsChecked();
+  }
+
+  ['@test with static values'](assert) {
+    this.render(`{{input type="checkbox" disabled=false tabindex=10 name="original-name" checked=false}}`);
+
+    this.assertSingleCheckbox();
+    this.assertCheckboxIsNotChecked();
+    this.assertNotDisabled();
+    this.assertAttr('tabindex', '10');
+    this.assertAttr('name', 'original-name');
+
+    this.runTask(() => this.rerender());
+
+    this.assertSingleCheckbox();
+    this.assertCheckboxIsNotChecked();
+    this.assertNotDisabled();
+    this.assertAttr('tabindex', '10');
+    this.assertAttr('name', 'original-name');
+  }
+
+});
+
+//TODO: GJ: decide what to do with the inconsistencies below
+// 1. the `size` attribute can be initially set to `null`, then updated to a value
+//    but then raises an exception when set to `null` again (perhaps this is intended?)
+// 2. the `placeholder` and `name` can be initially set to `null` and will have a value of `undefined`.
+//    they can then be set to a string, but when set to `null` again they will have a value of `null`.
+moduleFor(`@htmlbars Helpers test: {{input type='text'}}`, class extends InputRenderingTest {
```

Comment:
@chancancode I tried to be more comprehensive and test the behaviour of all attributes with `null` values, but I'm seeing inconsistencies with 1) the `size` and 2) the `placeholder` and `name` attributes:

  1. the `size` attribute can be initially set to `null`, then updated to a value but then raises an exception when set to `null` again (perhaps this is intended?)
  2. the `placeholder` and `name` can be initially set to `null` and will have a value of `undefined`. They can then be set to a string, but when set to `null` again they will have a value of `null`.

I've commented out the failing test assertions below. Perhaps you might know what the behaviour should be?